test/tokenizer/index.spec.ts

```diff
@@ -376,6 +376,262 @@ describe('scan', () => {
           },
         ].forEach(({ actual, expected }) => expect(actual).to.eql(expected));
       });
+
+      describe('custom parameters', () => {
+        it('should allow positional parameters for all dialects', () => {
```

Comment:
For all the tests here, should move the foreach over dialects (and any inner foreachs) up out of the test, where each dialect is its own testcase, where it's useful to know if one particular dialect was broken or all of them with a given change.